### ISSUE TITLE
Better pagination

### ISSUE THREAD

**Issue by dszlachta** [TrueBlocks/trueblocks-key#129]
### Keyset pagination
We need to change the way pagination works to avoid using SQL's `OFFSET` (it will kill the database if it's a large number).
Instead, we can use a pattern called _keyset pagination_. Every request tells us "what has been seen" and we can "chop off" that part of dataset.

To implement _keyset pagination_, we can use appearances. Request for the next page should include the last appearance from the current page, so we can fetch appearances earlier than the one sent in, limited in number by `pageSize`. 

Following this logic a request for the previous page would include the first appearance from the current page. This case is special, because asking for "appearances later than X" would return dataset starting at the latest block, i.e. invalid results. But we can do the query using reversed index (ascending ordering) and sort results in descending order before returning from the DB. It's cheap and takes the same time as next page query.

### Sorting order and `lastBlock`

Sorting is done in descending order, i.e. from the latest appearance to the earliest.

The user can set `lastBlock` parameter if they want to narrow the dataset:
- `lastBlock: 1000, pageSize: 100` gets 100 appearances starting from block `1000` (inclusive) 
- `lastBlock:"latest", pageSize: 100` gets 100 appearances starting from the latest (at that time) block (inclusive)
No `lastBlock` is same as `"latest"`.

### `pageId`

In order to generate a page of data in a stateless way, we need this information:
- `lastBlock` (stabilizes the first page)
- `direction`: do we want to fetch the previous or next page?
- `appearance`: in case of previous page ID, it will be the first appearance from the current page. For next page ID, it will be the last one.

To ease copying it (and limit noise in the RPC query) we can encode it, so it's a single string. We can either use hex or base64 encoding (no external packages needed): 
```json
{
"nextPageId": "ASbDVgEBJsMNAAAABw=="
}
``` 
Here's the code: https://go.dev/play/p/UcIk134G7q0

To sum up, every request returns, apart from data, 2 pageIds: for the previous and next page. The user is responsible for sending the right pageId with the subsequent request. 
No pageId means "first page".

### Minimal safe `pageSize`

Since adding `limit 1` to SQL queries can make them slow, I suggest setting minimal safe page size to `10`. So the user cannot fetch less than 10 items.
